src/url-state-machine.js

```diff
@@ -1419,66 +1107,86 @@ function urlToUnicode(domain) {
   }
 }
 
-function init(url, base) {
-  /*jshint validthis:true */
-  if (this === undefined) {
-    throw new TypeError("Failed to construct 'URL': Please use the 'new' operator, " +
-      "this DOM object constructor cannot be called as a function.");
-  }
-  if (arguments.length === 0) {
-    throw new TypeError("Failed to construct 'URL': 1 argument required, but only 0 present.");
-  }
+module.exports.createURLConstructor = function () {
+  // TODO
```

Comment:
We want to be sure someone can't overwrite objects for all contexts though - although that's a more general jsdom problem (which we might have the opportunity to solve with webidl2js while we're at it).